packages/logger/src/formats.ts

```diff
@@ -0,0 +1,130 @@
+import { TransformableInfo } from 'logform';
+import stringify from 'json-stringify-safe';
+import { EOL } from 'os';
+import { format } from 'winston';
+
+const errorParamPos = 0;
+
+const LEVEL_EMOJI: Record<string, string> = {
+  info: '🍺',
+  warn: '❗️',
+  error: '🔥',
+  default: '🤷‍♂️',
+};
+
+/**
+ * Returns emoji format based on level.
+ */
+const emojiLevelFormat = format(
+  (info: TransformableInfo): TransformableInfo => {
+    const { level } = info;
+    const emoji = LEVEL_EMOJI[level] || LEVEL_EMOJI.default;
+    return { ...info, level: `${emoji} ${level}` };
+  },
+);
+
+/**
+ * Extracts parameters from winston.
+ */
+const getWinstonParams = (info: TransformableInfo): any[] | undefined => {
+  // Winston adds all parameters to 'splat' property which is accessible only by Symbol[splat]
+  return info[Symbol.for('splat') as any];
+};
+
+/**
+ * Returns stringified parameters.
+ * Returns undefined if parameters are undefined.
+ * If `logFormat` is `json`, Error object is omitted since it'll be included as `message`.
+ * If `logFormat` is `simple`, `EOL` is printed at the start of each string.
+ */
+const stringifyParams = (params: any[] | undefined, logFormat: 'simple' | 'json'): string[] | undefined => {
+  if (params === undefined) {
+    return params;
+  }
+
+  switch (logFormat) {
+    case 'json':
+      // Remove Error object if it's present where it's expected to be & stringify
+      if (params[errorParamPos] instanceof Error) {
+        return [
+          ...params.slice(0, errorParamPos),
+          ...params.slice(errorParamPos + 1),
+        ].map(p => stringify(p));
+      }
+
+      return params.map(p => stringify(p));
+
+    default:
+      return params.map((v: any) => `${EOL}${stringify(v)}`);
+  }
+};
+
+/**
+ * Format log entry in a GCP compatible format.
+ */
+const gcpLogEntryFormat = format((info: TransformableInfo): TransformableInfo => {
```

Comment:
@knor-el-snor I believe we borrowed the rule from [AirBnB's styleguide](https://github.com/airbnb/javascript#naming--Acronyms-and-Initialisms) which states:

> Acronyms and initialisms should always be all uppercased, or all lowercased.

I think it should be lowercased for consistency sake. It is all lowercased & it is consistent with other variables in the file like `jsonFormat`

Reply:
@gorjuspixels We should definitely revisit this rule. I'll be updating our eslint a bit this week, will take it into consideration since we had the same discussion about `ID` a while back in Core team